### PR TITLE
Add local confirmation resend

### PR DESCRIPTION
**Pull request by KedziaPawel**
## What does this PR do?

Add resend flow for confirming email locally

#### *Any scout cleaning/refactoring?*
Removed setNotificationEmail as I didn't set any email and changed it to verifyNotificationEmail

- [ ] Checked on iOS
- [X] Checked on Android

### CODE REVIEW

File: error/messages.ts [hunk 1]
```diff
@@ -3,4 +3,5 @@ export default {
   txnMempoolConflictCode18: 'txn-mempool-conflict (code 18)',
   missingInputs: 'Missing inputs',
   dustCode64: 'dust (code 64)',
+  requestFailed5XX: 'Request failed with status code 5',
```
Comment (marek-siemieniuk-morawski) [inline error/messages.ts:6]: 5? Shouldn't be 5xx or someting?

File: src/components/TimeoutButton.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  };
+
+  return (
+    <FlatButton
+      testID={testID}
+      disabled={seconds > 0}
+      titleStyle={styles.titleStyle}
+      title={`${title}${getTimeout()}`}
+      onPress={onButtonPress}
+      containerStyle={containerStyle}
```
Comment (marek-siemieniuk-morawski) [inline src/components/TimeoutButton.tsx:46]: Shouldn't the button become disabled when the timer is clocking?

File: src/screens/LocalConfirmNotificationCodeScreen.tsx [hunk 3]
```diff
@@ -4,57 +4,68 @@ import React, { PureComponent } from 'react';
...
-  resendCode = () => {
-    this.setState({
-      error: '',
-      userCode: '',
-      numberAttempt: 0,
+  resendCode = (error = '') => {
+    const { verifyNotificationEmail } = this.props;
+    const { email } = this.props.route.params;
+
+    verifyNotificationEmail(email, {
```
Comment (marek-siemieniuk-morawski) [inline src/screens/LocalConfirmNotificationCodeScreen.tsx:49]: It's actually not about notification email, but just about verifying email. The endpoint is only to verify whether the user passed a valid email address that he has access to. So, I'd rename it but it's just a detail

File: error/messages.ts (hunk 1 above)
Comment (KedziaPawel) [reply error/messages.ts:6]: I was thinking about it and if I would put XX on the end I would have to slice those two last characters in the place of usage, to be honest, I'm not sure what is better

File: src/components/TimeoutButton.tsx (hunk 2 above)
Comment (KedziaPawel) [reply src/components/TimeoutButton.tsx:46]: it is, check the line 42

File: error/messages.ts (hunk 1 above)
Comment (marek-siemieniuk-morawski) [reply error/messages.ts:6]: Me either, I just thought you made a typo. Probably we don't want to inform users about status code at all - simply a technical error occured, please try again

File: src/screens/LocalConfirmNotificationCodeScreen.tsx (hunk 3 above)
Comment (KedziaPawel) [reply src/screens/LocalConfirmNotificationCodeScreen.tsx:49]: it depends how you read it, verify notification email means verify email for notification, we could have an email for a different feature for example verify airdrop email (of course we won't have, just an example)

File: src/components/TimeoutButton.tsx (hunk 2 above)
Comment (marek-siemieniuk-morawski) [reply src/components/TimeoutButton.tsx:46]: lol, I haven't spotted it. Thanks

File: loc/en.js [hunk 4]
```diff
@@ -239,6 +239,7 @@ module.exports = {
     goToWalletDetails: 'Go to Wallet Details',
   },
   notifications: {
+    resendCode: 'Resend code',
```
Comment (Kamill90) [inline loc/en.js:242]: already exists under key 'resend'

File: error/messages.ts (hunk 1 above)
Comment (KedziaPawel) [reply error/messages.ts:6]: no, I didn't make I typo. Yes that's why we catch all 5XX errors and translate to a general user-friendly message

File: loc/en.js (hunk 4 above)
Comment (KedziaPawel) [reply loc/en.js]: done

File: src/state/notifications/selectors.ts [hunk 5]
```diff
@@ -15,6 +18,13 @@ export const sessionToken = createSelector(local, state => state.sessionToken);
 export const notificationError = createSelector(local, state => state.error);
 export const storedEmail = createSelector(local, state => state.email);
 export const storedPin = createSelector(local, state => state.pin);
+export const error = createSelector(local, state => state.error);
+export const enhancedError = createSelector(error, err => {
```
Comment (Kamill90) [inline src/state/notifications/selectors.ts:22]: Very strange name for error. ReadableError? IDK, minor. There is `notificationError` above, no need to duplicate.

**Review by Kamill90 — APPROVED** [review]

File: src/state/notifications/selectors.ts (hunk 5 above)
Comment (KedziaPawel) [reply src/state/notifications/selectors.ts:22]: I like that, I'll rename

**Pull request merged**